parser.py

```diff
@@ -36,7 +39,20 @@ def parse(toParse):
     parsedString = dicePat.search(toParse)
     diceNo, sides = parsedString.groups()
     print("Rolling " + diceNo + ' dice with ' + sides + ' sides each.')
-    print("Result: " + str(roll(diceNo, sides)))
+    # print("Result: " + str(roll(diceNo, sides)))
+
+    sides = int(sides)
+    diceNo = int(diceNo)
+
+    diceList = [
+        dice.Dice([i + 1 for i in range(sides)])
+        for i in range(diceNo)
+    ]
+
+    dicePool = pool.DicePool(diceList)
+    result = dicePool.roll()
+    print (str(sum(result)) + ": " + str(result))
```

Comment:
Using a formatted string here would probably read nicer. I think something like this would work (although there's multiple specific ways of doing it, with either formatting syntax, depending on preference):
```
print('{sum}: {result}'.format(sum=sum(result), result=result))
```